drivers/hmis/app/models/hmis/hud/processors/enrollment_processor.rb

```diff
@@ -81,7 +81,8 @@ def assign_metadata
       return if active_unit_occupancy&.unit_id == unit.id
 
       # If assigned to a different unit: unassign
-      enrollment.active_unit_occupancy&.assign_attributes(occupancy_period_attributes: { end_date: Date.current })
+      # Update the occupancy period directly, instead of using enrollment.release_unit!, to avoid creating a nested transaction.
```

Comment:
Question: Why does the form processor's `assign_unit` functionality silently release the previously assigned unit? Whereas, other places we call the _enrollment's_ `assign_unit` function, which raises if the enrollment is already assigned to another unit.

Reply:
The Enrollment processor "current_unit" value is used in forms that change an enrollment's assigned unit. If you are changing the assigned unit, the previous unit should be released, which is why this behavior is here.
I'll drop a link to https://github.com/open-path/Green-River/issues/8505 since its related and I just fleshed it out a bit more